src/shared/udev-util.c

```diff
@@ -324,6 +324,26 @@ bool device_for_action(sd_device *dev, DeviceAction action) {
         return a == action;
 }
 
+int log_device_uevent(sd_device *device, int level, const char *str) {
```

Comment:
our log functions typically don't fail (or if they fail we mostly ignore that). And I think that makes a lot of sense: logging should be mostly side-effect free, it should document code flow, not change code flow.

Hence maybe rework this to not propagate errors but instead to include any failures in the log message generated by this, so that we can just call this function, and it does the right thing and we can forget about the rest?